airflow/contrib/hooks/gcs_hook.py

```diff
@@ -287,57 +236,41 @@ def is_updated_after(self, bucket, object, ts):
         :param ts: The timestamp to check against.
         :type ts: datetime.datetime
         """
-        service = self.get_conn()
-        try:
-            response = (service
-                        .objects()
-                        .get(bucket=bucket, object=object)
-                        .execute())
-
-            if 'updated' in response:
-                import dateutil.parser
-                import dateutil.tz
+        client = self.get_conn()
+        bucket = storage.Bucket(client=client, name=bucket)
+        blob = bucket.get_blob(blob_name=object)
+        blob.reload()
 
-                if not ts.tzinfo:
-                    ts = ts.replace(tzinfo=dateutil.tz.tzutc())
+        blob_update_time = blob.updated
 
-                updated = dateutil.parser.parse(response['updated'])
-                self.log.info("Verify object date: %s > %s", updated, ts)
+        if blob_update_time is not None:
+            import dateutil.tz
 
-                if updated > ts:
-                    return True
+            if not ts.tzinfo:
+                ts = ts.replace(tzinfo=dateutil.tz.tzutc())
 
-        except HttpError as ex:
-            if ex.resp['status'] != '404':
-                raise
+            self.log.info("Verify object date: %s > %s", blob_update_time, ts)
 
-        return False
+            if blob_update_time > ts:
+                return True
+            else:
+                return False
```

Comment:
'return False' is missing for if blob_udate_time is None

Reply:
Good call.